test/functional/lua/buffer_updates_spec.lua

```diff
@@ -438,6 +438,20 @@ describe('lua: nvim_buf_attach on_bytes', function()
         { "test1", "bytes", 1, 4, 1, 0, 1, 3, 0, 48, 0, 0, 0 };
       }
     end)
+
+    it("visual charwise paste", function()
+      local check_events = setup_eventcheck(verify, {'1234567890'})
+      funcs.setreg('a', '___')
+
+      feed '1G1|vll'
+      check_events {}
+
+      feed '"ap'
+      check_events {
+        { "test1", "bytes", 1, 3, 0, 0, 0, 0, 3, 3, 0, 0, 0 };
```

Comment:
Note: this should ideally be collapsed into one event (to save some bandwidth). There is nothing "wrong" with the current events semantically though.